api/solarperformanceinsight_api/compute.py

```diff
@@ -369,4 +377,52 @@ def run_performance_job(job: models.StoredJob, si: storage.StorageInterface):
             ),
         ]
     )
+    return monthly_energy, result_list
+
+
+def run_performance_job(job: models.StoredJob, si: storage.StorageInterface):
+    montlhy_energy, result_list = _calculate_performance(job, si)
+    save_results_to_db(job.object_id, result_list, si)
+
+
+def compare_expected_and_actual(job: models.StoredJob, si: storage.StorageInterface):
+    monthly_energy, result_list = _calculate_performance(job, si)
+    # performance granularity validation means there won't be system level
+    # perfromance and inverter level that you wouldn't want to sum
+    performance_data_ids = [
+        do.object_id
+        for do in job.data_objects
+        if do.definition.type == models.JobDataTypeEnum.actual_performance
+    ]
+    months = (
+        (job.definition.parameters.time_parameters._time_range)
+        .month.unique()  # type: ignore
+        .sort_values()
+    )
+    expected = monthly_energy["performance"]  # type: ignore
+    actual_performance = sum(
+        _get_data(job.object_id, did, si) for did in performance_data_ids
+    )[
+        "performance"
+    ]  # type: ignore
+    actual_energy = actual_performance.resample("1h").mean()  # type: ignore
+    actual_monthly_energy = (
+        actual_energy.groupby(actual_energy.index.month).sum().reindex(months)
+    )
+    diff = actual_monthly_energy - expected
+    ratio = actual_monthly_energy / expected
+    comparison_summary = pd.DataFrame(
+        {
+            "actual_energy": actual_monthly_energy,
+            "expected_energy": expected,
+            "difference": diff,
+            "ratio": ratio,
+        }
```

Comment:
@ncroft-b4 Is this what you had in mind for the output for use case 2C (in addition to the expected performance timeseries and weather timeseries)?

Reply:
Yes, that covers it. The monthly sum of insolation is also of interest, but it is not as important and its probably good to avoid displaying too much information on the output, and that can be pulled from the timeseries. So I say go with this